lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -412,20 +411,18 @@ defmodule RealtimeWeb.RealtimeChannel do
   def handle_in("access_token", %{"access_token" => refresh_token}, socket) when is_binary(refresh_token) do
     %{
       assigns: %{
-        access_token: access_token,
         pg_sub_ref: pg_sub_ref,
         channel_name: channel_name,
-        pg_change_params: pg_change_params,
-        tenant: tenant
+        pg_change_params: pg_change_params
       }
     } = socket
 
-    socket = assign(socket, :access_token, refresh_token)
+    # Update token and reset policies
+    socket = assign(socket, %{access_token: refresh_token, policies: nil})
 
     with {:ok, claims, confirm_token_ref, _, socket} <- confirm_token(socket),
```

Comment:
`confirm_token` was also calling `maybe_assign_policies` but we already call it in the next line. The confirm_token one was just resetting the policies due to the bug on `topic` assigns (see `confirm_token` comment)